### ISSUE TITLE
img-uses-alt does not understand object spread

### ISSUE THREAD

**Issue by lencioni** [evcohen/eslint-plugin-jsx-a11y#7]
This rule considers the following JSX bad:

```jsx
function Foo() {
  const props = { alt: 'foo' };

  return (
    <img {...props} />
  );
}
```

**Comment by evcohen** [comment]
Yeah, this is tough because we can't statically destructure spread object to find alt prop - it's documented [here](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/img-uses-alt.md)

However, maybe what we can do is allow an option in rule declaration in eslintrc that permits pass on spread.

**Comment by lencioni** [comment]
There might be some cases, such as the example I posted, where you could easily find the object definition and inspect its properties, right?

**Comment by robcolburn** [comment]
`Foo` looks like an image wrapper ( https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/img-has-alt.md#rule-details )

If so, I think: 

1. Do a `eslint-disable jsx-a11y/img-uses-alt` on Foo.js
2. Add `Foo` to your config as an image wrapper `"jsx-a11y/img-has-alt": [ 2, "Foo" ]`

Is that right?

**Comment by evcohen** [comment]
@robcolburn Yes, but I would destructure the props and be explicit in the wrapper component:

```js
function Foo({ alt, ...props}) {
    return <img alt={alt} {...props} />
}
```

**Comment by robcolburn** [comment]
@evcohen, it's a good enforcement technique too since you can make `alt` required. 👍

**Comment by evcohen** [comment]
@lencioni to answer your original question, we do have access to the current scope on the `context` object and can potentially handle the case that you mentioned. i'm hesitant to go in this direction, though because the cost of explicitly defining the `alt` prop is minimal and i think there is a benefit in terms of readability. 

can you think of a use-case like the one you provided where we aren't receiving props as an argument and declaring them in the function scope?

**Comment by lencioni** [comment]
> i'm hesitant to go in this direction, though because the cost of explicitly defining the alt prop is minimal and i think there is a benefit in terms of readability.

I agree with this. I think if you add your example from https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/7#issuecomment-216389401 to the documentation, we can close this issue.
